src/StatementProvider.js

```diff
@@ -0,0 +1,69 @@
+( function ( wb, $ ) {
+	'use strict';
+
+	/**
+	 * Combination of a claim, a rank and references.
+	 * @mixin wikibase.datamodel.StatementProvider
+	 *
+	 * @param {wikibase.datamodel.StatementGroupSet|null} [statementGroupSet=new wikibase.datamodel.StatementGroupSet()]
+	 * @license GPL-2.0+
+	 */
+	var SELF = wb.datamodel.StatementProvider = function WbDataModelStatementProvider( statementGroupSet ) {
+		statementGroupSet = statementGroupSet || new wb.datamodel.StatementGroupSet();
+
+		if( !( statementGroupSet instanceof wb.datamodel.StatementGroupSet ) ) {
+			throw new Error( 'Given argument `statementGroupSet` is not instance of `wb.datamodel.StatementGroupSet`' );
+		}
+		this._statementGroupSet = statementGroupSet;
+	};
+
+	/**
+	 * @class wikibase.datamodel.Statement
+	 */
+	$.extend( SELF.prototype, {
+		/**
+		 * @property {wikibase.datamodel.StatementGroupSet}
+		 * @protected
+		 */
+		_statementGroupSet: null,
+
+		/**
+		 * Returns all statements of given entity. Statements on sub-entities are NOT included.
+		 *
+		 * @return {wikibase.datamodel.StatementGroupSet}
+		 */
+		getStatements: function () {
+			return this._statementGroupSet;
+		},
+
+		/**
+		 * Finds a statement with given GUID either on current entity or on one of sub-entities.
+		 *
+		 * @param {string} guid
+		 *
+		 * @return {wikibase.datamodel.Statement|null} Statement that has given GUID. Returned statement can belong to the entity itself as
+		 *  well as to one of the sub-entities. `null` if entity is not found.
+		 */
+		findStatementByGuid: function (guid) {
+			var res = null;
+			this._statementGroupSet.each( function() {
+				this.getItemContainer().each( function() {
+					if ( this.getClaim().getGuid() === guid ) {
+						res = this;
```

Comment:
I've created the alternative with loops, but it makes it quiet ugly:
```js
findStatementByGuid: function (guid) {
	var result = null;
	var statementGroupSet = this._statementGroupSet.toArray();

	outer_loop:
		for ( var i = 0; i < statementGroupSet.length; i++) {
			var statementList = statementGroupSet[i].getItemContainer().toArray();
			for (var j = 0; j < statementList.length; j++) {
				var statement = statementList[j];
				if (statement.getClaim().getGuid() === guid) {
					result = statement;
					break outer_loop;
				}
			}
		}

	return result;
}
```
And, speaking about performance it won't be much faster as soon as it first traverses the `Set` to create array and then traverses the array. Don't think it is worth it.